lint/src/main/java/net/grandcentrix/thirtyinch/MissingTiViewImplementationDetector.java

```diff
@@ -0,0 +1,250 @@
+package net.grandcentrix.thirtyinch;
+
+import com.android.tools.lint.detector.api.Category;
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Implementation;
+import com.android.tools.lint.detector.api.Issue;
+import com.android.tools.lint.detector.api.JavaContext;
+import com.android.tools.lint.detector.api.Scope;
+import com.android.tools.lint.detector.api.Severity;
+import com.android.tools.lint.detector.api.TextFormat;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiClassType;
+import com.intellij.psi.PsiJavaCodeReferenceElement;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiType;
+import com.intellij.psi.PsiTypeParameter;
+import com.intellij.psi.util.PsiUtil;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.uast.UBlockExpression;
+import org.jetbrains.uast.UCallExpression;
+import org.jetbrains.uast.UClass;
+import org.jetbrains.uast.UExpression;
+import org.jetbrains.uast.UReferenceExpression;
+import org.jetbrains.uast.UastContext;
+import org.jetbrains.uast.UastUtils;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+
+@SuppressWarnings("WeakerAccess")
+public final class MissingTiViewImplementationDetector extends Detector implements Detector.UastScanner {
+
+    private static final String TI_VIEW_FQ = "net.grandcentrix.thirtyinch.TiView";
+    private static final String PROVIDE_VIEW_METHOD = "provideView";
+    private static final List<String> TI_CLASS_NAMES = unmodifiableList(asList(
+            "net.grandcentrix.thirtyinch.TiActivity",
+            "net.grandcentrix.thirtyinch.TiFragment"));
+    private static final String ADD_PLUGIN_METHOD = "addPlugin";
+    private static final String TI_ACTIVITY_PLUGIN_NAME = "TiActivityPlugin";
+    private static final String TI_FRAGMENT_PLUGIN_NAME = "TiFragmentPlugin";
+    private static final List<String> CA_CLASS_NAMES = unmodifiableList(asList(
+            "com.pascalwelsch.compositeandroid.activity.CompositeActivity",
+            "com.pascalwelsch.compositeandroid.fragment.CompositeFragment"));
+
+    static final Issue ISSUE = Issue.create(
```

Comment:
This should probably be split into two separate `Issue` descriptions with the same ID, which provide distinct descriptions for `Ti` and `Composite` use cases. Otherwise, the part about `provideView()` is just confusing.